File: sample/dotnet/BestsellerApi.ApiService/Entities/Tracking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BestsellerApi.ApiService.Entities;

[Table("tracking")]
public class Tracking
{
    [Key]
    [Column("tracking_no")]
    public string TrackingNo { get; set; } = string.Empty;

    [Required]
    [Column("status")]
    public string Status { get; set; } = string.Empty;

    [Required]
    [Column("current_location")]
    public string CurrentLocation { get; set; } = string.Empty;

    [Column("estimated_delivery")]
    public DateTimeOffset? EstimatedDelivery { get; set; }

    [Column("delivery_date")]
    public DateTimeOffset? DeliveryDate { get; set; }

    [Column("created_at")]
    public DateTimeOffset CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTimeOffset UpdatedAt { get; set; }

    public ICollection<TrackingEventEntity> Events { get; set; } = new List<TrackingEventEntity>();
}

File: sample/java/src/main/java/com/bestseller/api/entity/Tracking.java
package com.bestseller.api.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;

@Table("tracking")
public record Tracking(
    @Id
    @Column("tracking_no")
    String trackingNo,

    @Column("status")
    String status,

    @Column("current_location")
    String currentLocation,

    @Column("estimated_delivery")
    Instant estimatedDelivery,

    @Column("delivery_date")
    Instant deliveryDate,

    @Column("created_at")
    Instant createdAt,

    @Column("updated_at")
    Instant updatedAt
) {
}
